bin/publish-npm-modules.sh

```diff
@@ -63,15 +78,27 @@ mkdir -p "$SOCKET_HOME"
 export SOCKET_HOME
 export PREFIX
 
-if (( do_global_link && !dry_run )); then
-  echo >&2 "warning: '--link' implies '--dry-run' too"
-  dry_run=1
+# Confirm that ssc in PATH matches current commit
```

Comment:
While I like the intent here, I am not a fan of this type of check existing. There are edge cases to consider, like an already installed version of `ssc` that can be resolved within your `PATH` that takes precedence over the one installed from this script.

Reply:
As above, my intention is to ensure a sane package before publishing.

It's a bit of extra work to go through but I think it's well worth it.

What if instead of checking the `PATH` version, we check the instance that is copied to the npm staging folder?

Reply:
I'm not sure on this one. Maybe @heapwolf or @chicoxyzzy has an idea about this. I've never done a check like this before

Reply:
I think that instead of comparing with the `PATH` version, we should do `ci_version_check.{sh|ps1}` sanity check on CI when we will have a build+publish action there.

Reply:
(we already do this check after we build Socket)

Reply:
> I think that instead of comparing with the `PATH` version, we should do `ci_version_check.{sh|ps1}` sanity check on CI when we will have a build+publish action there.

That was my intention here @chicoxyzzy.
Think about a situation where some `.js` or `.cc` outside `ssc` changes, code is committed, but the binary doesn't change.
If we then run `publish-npm-packages`, the shipped `ssc` won't match the latest commit.
It would be less of a problem in a CI generated build because everything would be built from source in the one off CI environment.

Reply:
This one is still up for discussion. I have updated to test the package version rather than $PATH.